presto-main/src/main/java/com/facebook/presto/cost/HistoryBasedPlanStatisticsCalculator.java

```diff
@@ -201,7 +201,7 @@ private PlanNodeStatsEstimate getStatistics(PlanNode planNode, Session session,
                         Optional<HistoricalPlanStatisticsEntry> historicalPlanStatisticsEntry = getSelectedHistoricalPlanStatisticsEntry(entry.getValue(), inputTableStatistics.get(), historyMatchingThreshold);
                         if (historicalPlanStatisticsEntry.isPresent()) {
                             PlanStatistics predictedPlanStatistics = historicalPlanStatisticsEntry.get().getPlanStatistics();
-                            if (predictedPlanStatistics.getConfidence() > 0) {
+                            if (predictedPlanStatistics.getConfidence() > delegateStats.confidenceLevel().ordinal() - 1) {
```

Comment:
would be clearer to convert the predictedPlanStatistics to a confidenceLevel and then compare the confidence levels directly.  This kind of logic is very error prone, both because it's complex and hard to reason about, and because it relies on assumptions about the relationship between PlanStatistics.getConfidence() and ConfidenceLevel.ordinal() that aren't enforced anywhere and could change.

You should also then add a test for the conversion between confidence and confidenceLevel.